presto-main/src/main/java/com/facebook/presto/operator/repartition/ArrayBlockEncodingBuffer.java

```diff
@@ -201,17 +204,19 @@ public String toString()
     }
 
     @Override
-    protected void setupDecodedBlockAndMapPositions(DecodedBlockNode decodedBlockNode)
+    protected void setupDecodedBlockAndMapPositions(DecodedBlockNode decodedBlockNode, int partitionBufferCapacity)
     {
         requireNonNull(decodedBlockNode, "decodedBlockNode is null");
 
         decodedBlockNode = mapPositionsToNestedBlock(decodedBlockNode);
         ColumnarArray columnarArray = (ColumnarArray) decodedBlockNode.getDecodedBlock();
         decodedBlock = columnarArray.getNullCheckBlock();
 
+        setTargetBufferSize(decodedBlockNode, partitionBufferCapacity);
+
         populateNestedPositions(columnarArray);
 
-        ((AbstractBlockEncodingBuffer) valuesBuffers).setupDecodedBlockAndMapPositions(decodedBlockNode.getChildren().get(0));
+        ((AbstractBlockEncodingBuffer) valuesBuffers).setupDecodedBlockAndMapPositions(decodedBlockNode.getChildren().get(0), partitionBufferCapacity);
```

Comment:
Instead of casting in multiple places change the type of valuesBuffers to AbstractBlockEncodingBuffer

Reply:
Done in new commit `17ee832a6a Change nested buffers to AbstractBlockEncodingBuffer`